api/v1alpha1/nodepool_types.go

```diff
@@ -419,11 +419,74 @@ type PowerVSNodePoolPlatform struct {
 	ImageDeletePolicy string `json:"imageDeletePolicy,omitempty"`
 }
 
+// KubevirtCompute contains values associated with the virtual compute hardware requested for the VM.
+type KubevirtCompute struct {
+	// Memory represents how much guest memory the VM should have
+	//
+	// Defaults to 4Gi
+	//
+	// +optional
+	// +kubebuilder:default="4Gi"
+	Memory *resource.Quantity `json:"memory,omitempty"`
+
+	// Memory represents how many cores the guest VM should have
+	//
+	// Defaults to 2 cores
+	//
+	// +optional
+	// +kubebuilder:default=2
+	Cores *uint32 `json:"cores,omitempty"`
+}
+
+// KubevirtPersistentVolume containes the values involved with provisioning persistent storage for a KubeVirt VM.
+type KubevirtPersistentVolume struct {
+	// Size is the size of the persistent storage volume
+	//
+	// +optional
+	// +kubebuilder:default="16Gi"
+	Size *resource.Quantity `json:"size,omitempty"`
+	// StorageClass is the storageClass used for the underlying PVC that hosts the volume
+	//
+	// +optional
+	StorageClass *string `json:"storageClass,omitempty"`
+}
+
+// KubevirtRootVolume represents the volume that the rhcos disk will be stored and run from.
+type KubevirtRootVolume struct {
+	// Image represents what rhcos image to use for the node pool
+	Image *KubevirtDiskImage `json:"diskImage,omitempty"`
+
+	// KubevirtVolumeTypes represents of type of storage to run the image on
+	//
+	// +optional
+	KubevirtVolumeTypes `json:",inline"`
+}
+
+// KubevirtVolumeTypes represents what kind of storage to use for a KubeVirt VM volume
+type KubevirtVolumeTypes struct {
+	// Persistent volume type means the VM's storage is backed by a PVC
+	// VMs that use persistent volumes can survive disruption events like restart and eviction
+	// This is the default type used when no storage type is defined.
+	// +optional
+	Persistent *KubevirtPersistentVolume `json:"persistent,omitempty"`
+}
+
+// KubevirtDiskImage contains values representing where the rhcos image is located
+type KubevirtDiskImage struct {
+	// ContainerDiskImage is a string representing the container image that holds the root disk
+	ContainerDiskImage *string `json:"containerDiskImage,omitempty"`
+}
+
 // KubevirtNodePoolPlatform specifies the configuration of a NodePool when operating
 // on KubeVirt platform.
 type KubevirtNodePoolPlatform struct {
-	// NodeTemplate Spec contains the VirtualMachineInstance specification.
-	NodeTemplate *capikubevirt.VirtualMachineTemplateSpec `json:"nodeTemplate,omitempty"`
+	// RootVolume represents values associated with the VM volume that will host rhcos
+	RootVolume *KubevirtRootVolume `json:"rootVolume,omitempty"`
+
+	// Compute contains values representing the virtual hardware requested for the VM
+	//
+	// +optional
+	Compute KubevirtCompute `json:"compute,omitempty"`
```

Comment:
why is this not a pointer but RootVolume is? I think this field should actually be required, how can otherwise the backend work without this?